pkg/cluster/cluster_info_scraper.go

```diff
@@ -286,26 +287,31 @@ func (s *ClusterScraper) GetPVCs(namespace string, opts metav1.ListOptions) ([]*
 	return pvcs, nil
 }
 
-// GetPodGrandparentInfo gets grandParent (parent's parent) information of a pod: kind, name and uid
-// If pod's parent does not have parent, then return pod's parent info.
+// GetPodGrandparentInfo gets grandParent (parent's parent) information of a pod: kind, name, uid
+// If parent does not have parent, then return parent info.
 // Note: if parent kind is "ReplicaSet", then its parent's parent can be a "Deployment"
 //       or if its a "ReplicationController" its parent could be "DeploymentConfig" (as in openshift).
-func (s *ClusterScraper) GetPodGrandparentInfo(pod *api.Pod) (string, string, string, error) {
-	// Get pod controller info from cache if exists
+// The function also returns the retrieved parent and parents crud interface for use by the callers.
+func (s *ClusterScraper) GetPodGrandparentInfo(pod *api.Pod, ignoreCache bool) (string, string,
```

Comment:
I might be missing something here for the `ignoreCache` input argument. Looks all the places to call this `GetPodGrandparentInfo` func pass `ignoreCache` as false. And as I understand, the parent or grand parent info of a pod wouldn't be changed for the same pod with the same UID (the `podControllerInfoKey` is created from pod.UID). Maybe we could remove this argument?

Reply:
Thanks for catching this. This being false from all invocations is a miss. Actually the intention is to be able to retrieve the resources always, even if the cache exists in the case of volume based pod moves. The flag will be true [here](https://github.com/turbonomic/kubeturbo/pull/468/files#diff-623e1452732b5bf67c1efd4dd5ebb66aR246) now.